srcalc/src/main/java/gov/va/med/srcalc/domain/calculation/SignedResult.java

```diff
@@ -18,96 +27,257 @@
+    }
+    
+    /**
+     * As {@link #getCptCode()}, but represents a missing CPT Code as null. Purely to
+     * support Hibernate, which does not support Guava's Optional class.
+     * @return the optional CPT Code as a nullable string
+     */
+    @Column(name = "cpt_code", nullable = true, length = Procedure.CPT_CODE_LENGTH)
+    String getCptCodeNullable()
+    {
+        return fCptCode.orNull();
+    }
+    
+    /**
+     * For reflection-based construction only. Business code must provide this value to
+     * the constructor.
+     * @throws IllegalArgumentException if the cptCode is not {@link
+     * Procedure#CPT_CODE_LENGTH} characters
      */
-    public DateTime getStartDateTime()
+    void setCptCodeNullable(final String cptCode)
     {
-        return fStartDateTime;
+        setCptCode(Optional.fromNullable(cptCode));
     }
     
     /**
-     * The DateTime at which the user signed the calculation. We do not track
-     * milliseconds, so the millisOfSecond field will always be 0 to facilitate
-     * comparisons.
+     * Returns when the user signed the calculation. We do not track milliseconds, so this
+     * Date object will always represent a particular second on-the-dot.
      */
-    public DateTime getSignatureDateTime()
+    @Basic
+    @Column(nullable = false)
+    public Date getSignatureTimestamp()
     {
-        return fSignatureDateTime;
+        return fSignatureTimestamp;
+    }
+    
+    /**
+     * For reflection-based construction only. Business code must provide this value to
+     * the constructor.
+     */
+    void setSignatureTimestamp(final Date signatureTimestamp)
+    {
+        fSignatureTimestamp = Objects.requireNonNull(signatureTimestamp);
+    }
+    
+    /**
+     * Returns the calculation input values as a Map from variable key to value.
+     * @return an immutable map
+     */
+    @ElementCollection(fetch = FetchType.EAGER)
+    // Override various defaults for a better schema.
+    @CollectionTable(
+            name = "signed_result_input",
+            joinColumns = @JoinColumn(name = "result_id"))
+    @MapKeyColumn(name = "variable_key", nullable = false, length = Variable.KEY_MAX)
+    // There is no set maximum length for the value string, but the default of 255 should
+    // be sufficient.
+    @Column(name = "variable_value", nullable = false)
+    public Map<String, String> getInputs()
+    {
+        return fInputs;
+    }
+    
+    /**
+     * For reflection-based construction only. Business code must provide this value to
+     * the constructor.
+     */
+    void setInputs(final Map<String, String> inputs)
+    {
+        // Note: normally with Hibernate we would preserve the passed collection since it
+        // will be Hibernate's special implementation of Map. Since this object is
+        // immutable, however, we don't care to keep Hibernate's mutable implementation
+        // and just discard it.
+        fInputs = ImmutableMap.copyOf(inputs);
```

Comment:
Two things about the inputs and outcomes. What about checking for null as with the other variables? Also you are calling ImmutableMap.copyOf() in the constructor and that seems to be redundant since you also call the same thing here and with the setOutcomes() method.